common/remote/sockfile.cpp

```diff
@@ -2353,20 +2361,23 @@ class CRemoteFileIO : implements IFileIO, public CInterface
         // then also send sMode, cFlags
         unsigned short sMode = parent->getShareMode();
         unsigned short cFlags = parent->getCreateFlags();
-        switch ((compatIFSHmode)_compatmode)
+        if (!(parent->getIsShareSet()))
         {
-            case compatIFSHnone:
-                sMode = IFSHnone;
-                break;
-            case compatIFSHread:
-                sMode = IFSHread;
-                break;
-            case compatIFSHwrite:
-                sMode = IFSHfull;
-                break;
-            case compatIFSHall:
-                sMode = IFSHfull;
-                break;
+            switch ((compatIFSHmode)_compatmode)
+            {
+                case compatIFSHnone:
+                    sMode = IFSHnone;
+                    break;
+                case compatIFSHread:
+                    sMode = IFSHread;
+                    break;
+                case compatIFSHwrite:
+                    sMode = IFSHfull;
+                    break;
+                case compatIFSHall:
+                    sMode = IFSHfull;
+                    break;
+            }
```

Comment:
I can see this makes sense, in that it doesn't deduce share mode if explicitly set by setShareMode().

But it still feels that since sMode and _compatmode are serialized to the server, it should be the remote side (dafilesrv) that determines what to do with the compatibility flag.

Reply:
When a valid sMode is sent, dafilesrv uses that instead of _ccompatmode to set the share mode, precisely because it was explicitly set by the client so it should be honored by the server.
Otherwise _compatmode is a multi-purpose value that is used to sometimes set the share mode and sometimes set the create mode.